File: client/src/main/scala/stasis/client/ops/backup/Providers.scala
package stasis.client.ops.backup

import java.nio.file.FileSystem

import io.github.sndnv.layers.telemetry.TelemetryContext

import stasis.client.analysis.Checksum
import stasis.client.api.clients.Clients
import stasis.client.compression.Compression
import stasis.client.encryption
import stasis.client.staging.FileStaging
import stasis.client.tracking.BackupTracker

final case class Providers(
  checksum: Checksum,
  staging: FileStaging,
  compression: Compression,
  encryptor: encryption.Encoder,
  decryptor: encryption.Decoder,
  clients: Clients,
  track: BackupTracker,
  telemetry: TelemetryContext,
  filesystem: FileSystem
)

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/ops/backup/Providers.kt
package stasis.client_android.lib.ops.backup

import stasis.client_android.lib.encryption.Decoder as EncryptionDecoder
import stasis.client_android.lib.encryption.Encoder as EncryptionEncoder
import stasis.client_android.lib.analysis.Checksum
import stasis.client_android.lib.api.clients.Clients
import stasis.client_android.lib.compression.Compression
import stasis.client_android.lib.staging.FileStaging
import stasis.client_android.lib.tracking.BackupTracker

data class Providers(
    val checksum: Checksum,
    val staging: FileStaging,
    val compression: Compression,
    val encryptor: EncryptionEncoder,
    val decryptor: EncryptionDecoder,
    val clients: Clients,
    val track: BackupTracker
)
